src/archivematicaCommon/lib/elasticSearchFunctions.py

```diff
@@ -538,16 +538,16 @@ def index_mets_file_metadata(client, uuid, metsFilePath, index, type_, sipName,
     tree = ElementTree.parse(metsFilePath)
     root = tree.getroot()
 
+    # Map each element's parent because ElementTree doesn't
+    # provide a direct way to do it
+    parent_elements = {}
+    for parent in root.iter():
+        for child in parent:
+            parent_elements[child] = parent
+
```

Comment:
I wonder if the parents could be taken using the `..` XPath expression. It looks like it should work as long as the path doesn't attempt to reach the ancestors of the start element:

2.x: https://docs.python.org/2/library/xml.etree.elementtree.html#supported-xpath-syntax
3.x: https://docs.python.org/3/library/xml.etree.elementtree.html#supported-xpath-syntax

If that doesn't work, this mapping seems a bit excessive for this matter. We could iterate over the `mets:structMap[@TYPE='physical']` only and create a mapping with just the `FILEID -> DMDID` relations.

Reply:
I tried extracting only the physical struct map but had namespace problems when I tried to search inside of it. It looked like the current namespaces mapping only work off the root element. I'll make the change again and show you.

Reply:
No need for iterating the physical struct map since the `..` syntax worked!